packages/ember-engines/package.json

```diff
@@ -118,8 +118,9 @@
     "calculate-cache-key-for-tree": "^2.0.0",
     "ember-asset-loader": "^1.0.0",
     "ember-cli-babel": "^7.26.11",
+    "ember-cli-clean-css": "^3.0.0",
```

Comment:
@villander We are calling `preprocessMinifyCss` in [engine-addon.js](https://github.com/ember-engines/ember-engines/blob/3c70624a4bc458ea76fc7580f7a04fab4081a146/packages/ember-engines/lib/engine-addon.js#L766) which calls ember-cli-preprocess-registry to load registered `minify-css` plugins to minify. [Ref](https://github.com/ember-cli/ember-cli-preprocess-registry/blob/1d0ee06174a33c58ae2d99c656b0d127d87e4199/preprocessors.js#L108)

However, after upgrading ember-cli-preprocess-registry to v5.0.0, the minification does not happen because there is no registered plugins for minify-css [Ref](https://github.com/ember-cli/ember-cli-preprocess-registry/blob/5c2cc1865d1f72dbeffbee62fdb235cb7e83d30c/preprocessors.js#L89). This is a breaking change and that end users should include ember-cli-clean-css in their projects. [Changelog](https://github.com/ember-cli/ember-cli-preprocess-registry/releases/tag/v5.0.0)

There is an explicit test to ensure css is minified by ember-engines here https://github.com/ember-engines/ember-engines/blob/master/packages/ember-engines/node-tests/acceptance/build-test.js#L95C9-L95C13

I am bit of stuck here too. If ember-engines have to explicitly process css in such a way, do we need to load the minify-css plugin from ember-cli-clean-css ourselves ? 

Any pointers ? @void-mAlex @SergeAstapov

Reply:
@mukilane the package responsible for minify CSS into ember-engines is the `ember-cli-preprocess-registry` so if there's a bug about not minifying CSS it should be fixed there IMO.

Reply:
@villander Well, `ember-cli-preprocess-registry` was able to minify css so far was because it loaded a fallback preprocessors if there was none registered.

That is now removed with v5.0.0 as a breaking change, explicitly mentioning that minification will be handled by separate addon ( `ember-cli-clean-css` ).

`ember-cli-preprocess-registry` will only act as a registry and not have any default preprocessors registered.

`ember-cli` was also updated to include `ember-cli-clean-css` as a dependency in the blueprints and in the release blog as well.

Reply:
@mukilane thanks for refreshing my mind about it. Quick question, given the ember-cli already have this addon, isn't better just use it instead of adding this addon here? I'm wondering if we really need to add it since ember-cli already has this power.

Reply:
@villander I thought so too. And I have tried by adding `ember-cli-clean-css` in the consuming dummy app to test.

But the problem was that the registry to which the plugins are added is [instantiated once](https://github.com/ember-cli/ember-cli/blob/78aafc341111cb2065e6b42e176edebf28b1d8b7/lib/models/addon.js#L262C23-L262C37) per addon and one for the host app itself.

If I add clean css in `devDependencies` of the dummy app, only dummy's css are minified and not the engine's, because it is registered only to the dummy app.

If I add it as a dependency in the dummy app, then both are minified, since addon instantiates the packages in `dependencies`.

So at this point, the engine addons should include ember-cli-clean-css in their dependencies to ensure css minification. Engines could bring their choice of css minifier.

Alternatively, I tried to get the host's registry and use that for minification. This does solve the issue.

```js
let shouldMinifyCSS =
  'enabled' in this.options.minifyCSS
    ? this.options.minifyCSS.enabled
    : process.env.EMBER_ENV === 'production';
let minificationOptions = this.options.minifyCSS.options || {};
minificationOptions.registry = findLCAHost(this)?.registry || this.registry;
```
https://github.com/ember-engines/ember-engines/blob/3c70624a4bc458ea76fc7580f7a04fab4081a146/packages/ember-engines/lib/engine-addon.js#L722

Maybe we can go down this path ?

Reply:
> Maybe we can go down this path ?

Totally, that's the better way to handle that. You can create a conditional for some specific ember version if that breaks older versions.